tests/components/influxdb/test_init.py

```diff
@@ -11,748 +11,1116 @@
+async def test_setup_minimal_config(hass, mock_client, config_ext, get_write_api):
+    """Test the setup with minimal configuration and defaults."""
+    config = {"influxdb": {}}
+    config["influxdb"].update(config_ext)
+
+    assert await async_setup_component(hass, influxdb.DOMAIN, config)
+    assert hass.bus.listen.called
+    assert EVENT_STATE_CHANGED == hass.bus.listen.call_args_list[0][0][0]
+    assert get_write_api(mock_client).call_count == 1
+
+
+@pytest.mark.parametrize(
+    "mock_client, config_ext, get_write_api",
+    [
+        (influxdb.DEFAULT_API_VERSION, {"username": "user"}, _get_write_api_mock_v1),
+        (influxdb.DEFAULT_API_VERSION, {"token": "token"}, _get_write_api_mock_v1),
+        (
+            influxdb.API_VERSION_2,
+            {"api_version": influxdb.API_VERSION_2, "organization": "organization"},
+            _get_write_api_mock_v2,
+        ),
+        (
+            influxdb.API_VERSION_2,
+            {
+                "api_version": influxdb.API_VERSION_2,
+                "token": "token",
+                "organization": "organization",
                 "username": "user",
                 "password": "pass",
-                "exclude": {
-                    "entities": ["fake.blacklisted"],
-                    "domains": ["another_fake"],
-                },
-            }
+            },
+            _get_write_api_mock_v2,
+        ),
+    ],
+    indirect=["mock_client"],
+)
+# pylint: disable=redefined-outer-name
+async def test_invalid_config(hass, mock_client, config_ext, get_write_api):
+    """Test the setup with invalid config or config options specified for wrong version."""
+    config = {"influxdb": {}}
+    config["influxdb"].update(config_ext)
+
+    assert not await async_setup_component(hass, influxdb.DOMAIN, config)
+
+
+async def _setup(hass, mock_influx_client, config_ext, get_write_api):
+    """Prepare client for next test and return event handler method."""
+    config = {
+        "influxdb": {
+            "host": "host",
+            "exclude": {"entities": ["fake.blacklisted"], "domains": ["another_fake"]},
         }
-        config["influxdb"].update(kwargs)
-        assert setup_component(self.hass, influxdb.DOMAIN, config)
-        self.handler_method = self.hass.bus.listen.call_args_list[0][0][1]
-        mock_client.return_value.write_points.reset_mock()
-
-    def test_event_listener(self, mock_client):
-        """Test the event listener."""
-        self._setup(mock_client)
-
-        # map of HA State to valid influxdb [state, value] fields
-        valid = {
-            "1": [None, 1],
-            "1.0": [None, 1.0],
-            STATE_ON: [STATE_ON, 1],
-            STATE_OFF: [STATE_OFF, 0],
-            STATE_STANDBY: [STATE_STANDBY, None],
-            "foo": ["foo", None],
+    }
+    config["influxdb"].update(config_ext)
+    assert await async_setup_component(hass, influxdb.DOMAIN, config)
+    get_write_api(mock_influx_client).reset_mock()
```

Comment:
Mocks that are set up by a fixture is clean by default. All the fixtures here are scoped per test, so resetting the mock shouldn't be necessary.

Reply:
So the reason for this one is because there's a call to the write API during setup of the component to test if the config given for the connection is valid. Without this the tests with loops would have an issue since the call count would be 2 in the first loop and then 1 for each loop after that once reset.

How about instead I add a comment explaining that? Because you're right, it is confusing, it looks like a mistake.

Reply:
Ok, yes please add a comment.

Reply:
Ok made all requested changes. Let me know if adding a comment to explain this decision is ok or if you would prefer a different approach.